src/app/compliance/CompliancePage.js

```diff
@@ -126,10 +138,37 @@ class CompliancePage extends Component<Props, State> {
     isImportConfirmModalOpen: false,
     startDateAdded: null,
     endDateAdded: null,
+    isFreezeModalOpen: false,
+    isFrozenModalOpen: false,
   }
 
   componentWillMount () {
     this.props.fetchWhitelist()
+    this.props.getFreezeStatus().then(()=>{
+      if(this.props.isTokenFrozen){
+        this.setState({ isFrozenModalOpen: true })
+      }
+    })
+  }
+
+  handleFreezeSubmit = () => {
+    this.setState({ isFreezeModalOpen: true })
+  }
+
+  handleFreezeConfirm = () => {
+    this.setState({ isFreezeModalOpen: false })
+    this.props.toggleFreeze((()=>{
+      this.setState({ isFrozenModalOpen: true })
```

Comment:
put here TODO please which says that we shouldn't  extend ac code inside of the components and instead we should manage modal windows right in the ac